### PR TITLE
Feature exercice

### PR DESCRIPTION
**Pull request by KevinTss**
#32

### CODE REVIEW

File: src/controllers/lang/index.js [hunk 1]
```diff
@@ -1,32 +1,13 @@
...
-      })
-    }
-  },
-
   retrieveLangs: async (req, res) => {
     try {
       const langs = await read()
 
-      res.status(201).json(langs)
+      res.status(209).json(langs)
```
Comment (KevinTss) [inline src/controllers/lang/index.js:10]: Should be status 200

File: src/controllers/sport/handlers.js [hunk 2]
```diff
@@ -0,0 +1,82 @@
...
+  createSportContent,
+  createSport,
+  readSport,
+  readSportContent,
+} = require('./queries')
+
+module.exports = {
+  /**
+   * @param {object} data {name, langId}
+   * @return Created sport
```
Comment (KevinTss) [inline src/controllers/sport/handlers.js:15]:
```suggestion
   * @return {object} Created sport
```

File: src/controllers/sport/handlers.js [hunk 3]
```diff
@@ -0,0 +1,82 @@
...
+      name,
+    })
+
+    return { newSport, newSportContent }
+  },
+
+  /**
+   * @param {string} id Sport ID
+   * @param {string} lang Language ISO_639_1 (e.g. 'en')
+   * @return Sport
```
Comment (KevinTss) [inline src/controllers/sport/handlers.js:41]:
```suggestion
   * @return {object} Sport
```

File: src/controllers/sport/queries.js [hunk 4]
```diff
@@ -0,0 +1,50 @@
...
+    })
+
+    return newSport.save()
+  },
+
+  /**
+   * @param {object} data { sport, lang, name }
+   * @return Mongoose query object
+   */
+  createSportContent(data) {
```
Comment (KevinTss) [inline src/controllers/sport/queries.js:24]: Better to pass directly value instead of one object

File: src/controllers/sport/queries.js [hunk 5]
```diff
@@ -0,0 +1,50 @@
+'use strict'
+
+const mongoose = require('mongoose')
+
+const Sport = require('../../models/sport')
+const SportContent = require('../../models/sport/sport-content')
+
+module.exports = {
+  /**
+   * @return Mongoose query object
```
Comment (KevinTss) [inline src/controllers/sport/queries.js:10]:
```suggestion
   * @return {object} Mongoose query object
```

File: src/controllers/sport/queries.js [hunk 6]
```diff
@@ -0,0 +1,50 @@
...
+    const newSport = new Sport({
+      _id: new mongoose.Types.ObjectId(),
+    })
+
+    return newSport.save()
+  },
+
+  /**
+   * @param {object} data { sport, lang, name }
+   * @return Mongoose query object
```
Comment (KevinTss) [inline src/controllers/sport/queries.js:22]:
```suggestion
   * @return {object} Mongoose query object
```

File: src/controllers/sport/queries.js [hunk 7]
```diff
@@ -0,0 +1,50 @@
...
+      lang: new mongoose.Types.ObjectId(data.lang),
+      name: data.name,
+    })
+
+    return newSportContent.save()
+  },
+
+  /**
+   * @param {object} query Mongo query object
+   * @return Mongoose query object
```
Comment (KevinTss) [inline src/controllers/sport/queries.js:37]:
```suggestion
   * @return {object} Mongoose query object
```

File: src/controllers/sport/queries.js [hunk 8]
```diff
@@ -0,0 +1,50 @@
...
+   */
+  readSport(query = {}) {
+    return Sport.find(query)
+  },
+
+  /**
+   * @param {object} query Mongo query object
+   * @return Mongoose query object
```
Comment (KevinTss) [inline src/controllers/sport/queries.js:45]:
```suggestion
   * @return {object} Mongoose query object
```

File: src/controllers/user/handlers.js [hunk 9]
```diff
@@ -1,18 +1,35 @@
...
-  createUser: async (data) => {
-    const { email, firstName, lastName, password } = data
-
+  /**
+   * @param {string} email user email
+   * @param {string} firstName user first name
+   * @param {string} lastName user last name
+   * @param {string} phone user phone
+   * @param {string} password user password
+   * @param {boolean} isCoach If the user is a coach
```
Comment (alvermeylen) [inline src/controllers/user/handlers.js:17]: FLow ot be defined, but the way I see it and the way it is at the moment, the onboarding flow is the same for coaches and other users. Once logged in, you can "become a coach", then we would set the flag to true. A bit like airbnb. I think it makes things easier and is more user friendly. to be discussed.

Comment (KevinTss) [reply src/controllers/user/handlers.js:17]: @alvermeylen You're right, I'm for having different onboarding depend of each cases

**Pull request merged**